### ISSUE TITLE
custom/text module cannot click to execute 'chromium-browser &'

### ISSUE THREAD

**Issue by DrunkenChicken** [jaagr/polybar#1717]
## Describe the issue
With config:
[module/browser]
type = custom/text
content = 
content-padding = 1
content-foreground = ${color.fg-alt}
click-left = firefox &
click-right = chromium-browser &

firefox will start normally
but chromium will not show up.
result of `ps -ef | grep "chromium"` is empty.
It seem like chromium has been starting, but process got killed.

## To Reproduce
Add a text icon in the bar:
[module/browser]
type = custom/text
content = 
content-padding = 1
content-foreground = ${color.fg-alt}
click-left = firefox &
click-right = chromium-browser &

and just right click it.

## Polybar Log
This is part of log.
```
* Uncaught input event, forwarding to shell... (input: chromium-browser &)
* Executing shell command: chromium-browser &
- command: Waiting for pid 28087 to finish...
- command: Exited with status 0
```
It seems like the command has been executed. Below is the full log.

```
polybar -l=trace main
* Loading config: /home/brant/.config/polybar/config
- config: Current bar section: [bar/main]
* Loaded monitor DVI-I-0 (1920x1080+0+0)
* Bar geometry: 1920x22+0+0; Borders: 1,1,1,1
- bar: Attach X event sink
- bar: Attach signal receiver
- controller: Install signal handler
- controller: Setup user-defined modules
- module/battery: Deconstructing
error: Disabling module "battery" (reason: No suitable way to get current charge state)
* Starting application
- controller: Main thread id = 1
* Starting module/arch
* Starting module/term
* Starting module/files
- module/arch: Thread id = 2
* Starting module/browser
- module/term: Thread id = 3
* Starting module/snapshot
- module/files: Thread id = 4
* Starting module/settings
- module/browser: Thread id = 5
- module/snapshot: Thread id = 6
* Starting module/workspaces
- module/settings: Thread id = 7
* Starting module/volume
- module/workspaces: Thread id = 8
* Starting module/network
- module/volume: Thread id = 9
* Starting module/date
* Starting module/powermenu
* module/network: Invoking shell command: "~/.config/polybar/scripts/check-network"
- module/powermenu: Thread id = 10
* Eventqueue worker (thread-id=12)
- module/date: Thread id = 11
* Entering event loop (thread-id=1)
- bar: Create renderer
- renderer: Get TrueColor visual
- renderer: Allocate colormap
- renderer: Allocate output window
- renderer: Allocate window pixmaps
- renderer: Allocate graphic contexts
- renderer: Allocate alignment blocks
- renderer: Allocate cairo components
- renderer: Load fonts
* Configured DPI = 96x96
* Loaded font "YaHei Consolas Hybrid:size=12" (name=YaHei Consolas Hybrid, offset=2, file=/usr/share/fonts/truetype/YaHei Consolas Hybrid/YaHei Consolas Hybrid 1.12.ttf)
* Loaded font "Fantasque Sans Mono:pixelsize=10" (name=Fantasque Sans Mono, offset=2, file=/home/brant/.fonts/FantasqueSansMono-Regular.ttf)
* Loaded font "Material Icons:size=14" (name=Material Icons, offset=4, file=/home/brant/.fonts/MaterialIcons-Regular.ttf)
* Bar window: 0x4400001
- bar: Reconfigure window
- bar: Set window WM_NAME
- bar: Set window _NET_WM_WINDOW_TYPE
- bar: Set window _NET_WM_STATE
- bar: Set window _NET_WM_DESKTOP
- bar: Set window _NET_WM_PID
- bar: Map window
- bar: Draw empty bar
- bar: Setup tray manager
- tray: Find systray selection atom for the default screen
* Activating tray manager
- tray: Create tray window
* Tray window: 0x4400005
- tray: Restacking tray window
error: tray: Failed to put tray above 0x4400001 in the stack (XCB_MATCH (8))
- bar: Set window WM_NAME / WM_CLASS
- tray: Set window WM_PROTOCOLS
- tray: Set window _NET_WM_WINDOW_TYPE
- tray: Set window _NET_WM_STATE
- tray: Set window _NET_WM_PID
- tray: Set window _NET_SYSTEM_TRAY_VISUAL
- tray: Set window _NET_SYSTEM_TRAY_ORIENTATION
- tray: Set _NET_SYSTEM_TRAY_COLORS to ff1f1f1f
warn: Systray selection already managed (window=0x0c00005)
- tray: Listen for events on the new selection window
* Deactivating tray manager
- tray: Unembed clients
- tray: Destroy window
* module/arch: Rebuilding cache
* module/term: Rebuilding cache
* module/files: Rebuilding cache
* module/browser: Rebuilding cache
* module/snapshot: Rebuilding cache
* module/settings: Rebuilding cache
* module/workspaces: Rebuilding cache
* module/volume: Rebuilding cache
* module/network: Rebuilding cache
* module/date: Rebuilding cache
* module/powermenu: Rebuilding cache
- bar: Force update
* Redrawing bar window
- renderer: flush(1 geom=312x20+0+0, falloff=0)
- renderer: flush(2 geom=165x20+910+0, falloff=0)
- renderer: flush(3 geom=246x20+1672+0, falloff=0)
- bar: Received expose event
- bar: Force update
* Redrawing bar window
- renderer: flush(1 geom=312x20+0+0, falloff=0)
- renderer: flush(2 geom=165x20+910+0, falloff=0)
- renderer: flush(3 geom=246x20+1672+0, falloff=0)
* module/network: Rebuilding cache
* Redrawing bar window
- renderer: flush(1 geom=312x20+0+0, falloff=0)
- renderer: flush(2 geom=165x20+891+0, falloff=0)
- renderer: flush(3 geom=283x20+1635+0, falloff=0)
- bar: Detected motion: 0 at pos(239, 21)
- Found matching input area
- bar: Detected motion: 0 at pos(237, 21)
- Found matching input area
- bar: Detected motion: 0 at pos(234, 20)
- Found matching input area
- bar: Detected motion: 0 at pos(233, 20)
- Found matching input area
- bar: Detected motion: 0 at pos(232, 20)
- Found matching input area
- bar: Detected motion: 0 at pos(229, 20)
- Found matching input area
- bar: Detected motion: 0 at pos(228, 20)
- Found matching input area
- bar: Detected motion: 0 at pos(227, 20)
- Found matching input area
- bar: Detected motion: 0 at pos(224, 20)
- Found matching input area
- bar: Detected motion: 0 at pos(221, 19)
- Found matching input area
- bar: Detected motion: 0 at pos(219, 18)
- No matching cursor area found
- bar: Detected motion: 0 at pos(217, 18)
- bar: Detected motion: 0 at pos(212, 17)
- bar: Detected motion: 0 at pos(210, 16)
- bar: Detected motion: 0 at pos(209, 16)
- bar: Detected motion: 0 at pos(208, 16)
- bar: Detected motion: 0 at pos(207, 16)
- bar: Detected motion: 0 at pos(204, 15)
- bar: Detected motion: 0 at pos(203, 15)
- bar: Detected motion: 0 at pos(201, 13)
- Found matching input area
- bar: Detected motion: 0 at pos(199, 12)
- Found matching input area
- bar: Detected motion: 0 at pos(198, 12)
- Found matching input area
- bar: Detected motion: 0 at pos(197, 12)
- Found matching input area
- bar: Detected motion: 0 at pos(196, 11)
- Found matching input area
- bar: Detected motion: 0 at pos(194, 11)
- Found matching input area
- bar: Detected motion: 0 at pos(193, 10)
- Found matching input area
- bar: Detected motion: 0 at pos(191, 8)
- Found matching input area
- bar: Detected motion: 0 at pos(190, 7)
- Found matching input area
- bar: Detected motion: 0 at pos(189, 7)
- Found matching input area
- bar: Received button press: 3 at pos(189, 7)
- Found matching input area
* Uncaught input event, forwarding to shell... (input: chromium-browser &)
* Executing shell command: chromium-browser &
- command: Waiting for pid 28087 to finish...
- command: Exited with status 0
- bar: Force update
* Redrawing bar window
- renderer: flush(1 geom=312x20+0+0, falloff=0)
- renderer: flush(2 geom=165x20+891+0, falloff=0)
- renderer: flush(3 geom=283x20+1635+0, falloff=0)
- bar: Detected motion: 0 at pos(190, 9)
- Found matching input area
- bar: Detected motion: 0 at pos(193, 14)
- Found matching input area
- bar: Detected motion: 0 at pos(195, 16)
- Found matching input area
- bar: Detected motion: 0 at pos(196, 18)
- Found matching input area
- bar: Detected motion: 0 at pos(199, 21)
- Found matching input area
warn: Termination signal received, shutting down...
- controller: Uninstall sighandler
- controller: Detach signal receiver
- controller: Stop modules
* module/arch: Stopping
- module/arch: Release sleep lock
- module/arch: Deconstructing
* Deconstruction of module/arch took 0 ms.
* module/term: Stopping
- module/term: Release sleep lock
- module/term: Deconstructing
* Deconstruction of module/term took 0 ms.
* module/files: Stopping
- module/files: Release sleep lock
- module/files: Deconstructing
* Deconstruction of module/files took 0 ms.
* module/browser: Stopping
- module/browser: Release sleep lock
- module/browser: Deconstructing
* Deconstruction of module/browser took 0 ms.
* module/snapshot: Stopping
- module/snapshot: Release sleep lock
- module/snapshot: Deconstructing
* Deconstruction of module/snapshot took 0 ms.
* module/settings: Stopping
- module/settings: Release sleep lock
- module/settings: Deconstructing
* Deconstruction of module/settings took 0 ms.
* module/workspaces: Stopping
- module/workspaces: Release sleep lock
- module/workspaces: Deconstructing
* Deconstruction of module/workspaces took 0 ms.
* module/volume: Stopping
- module/volume: Release sleep lock
- module/volume: Deconstructing
* Deconstruction of module/volume took 2 ms.
- module/network: Release sleep lock
- command: Sending SIGTERM to running child process (28075)
- command: Waiting for pid 28075 to finish...
- command: killed by signal 15
* module/network: Stopping
- module/network: Release sleep lock
- module/network: Deconstructing
* Deconstruction of module/network took 23 ms.
* module/date: Stopping
- module/date: Release sleep lock
- module/date: Deconstructing
* Deconstruction of module/date took 0 ms.
* module/powermenu: Stopping
- module/powermenu: Release sleep lock
- module/powermenu: Deconstructing
* Deconstruction of module/powermenu took 0 ms.
- controller: Joining threads
* Waiting for spawned processes to end
* Reached end of application...

```
## Environment:
* WM: i3 version 4.16.1 (2019-01-27) © 2009 Michael Stapelberg and contributors
* Distro: Ubuntu 16.04 Linux brant 4.4.0-104-generic #127-Ubuntu SMP Mon Dec 11 12:16:42 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
* Output of `polybar -vvv`:
polybar 3.3.0-44-g82b9ea8

Features: +alsa +curl +i3 +mpd +network(libnl) +pulseaudio +xkeyboard

X extensions: +randr (-monitors) -render -damage -sync +composite +xkb +xrm +xcursor

Build type: RelWithDebInfo
Compiler: /usr/bin/g++
Compiler flags:  -Wall -Wextra -Werror -Wno-noexcept-type -pedantic -pedantic-errors
Linker flags: 

## Additional context
In utils/command.cpp, function exec, line 85,
If I add an usleep here, like:
```
 } else {
    usleep(1000000);
    // Close file descriptors that won't be used by the parent
    if ((m_stdin[PIPE_READ] = close(m_stdin[PIPE_READ])) == -1) {
      throw command_error("Failed to close fd");
  }
```
Then 'chromium-browser &' worked.

It's weird, so I look for where this function being executed. Then found in component/controller.cpp function process_inputdata()
I comment the line: 
```
\\m_command.reset();
```
And chromium-browser & worked without usleep() in command.cpp.

Maybe m_command has been released too quickly causing my problem.
But I don't know should this line being commented.
m_command is an unique_ptr, without reset(), it should also be auto released later, right?

Maybe it can also solve #860 #679

**Comment by patrick96** [comment]
I think this is the [Known Issue](https://github.com/jaagr/polybar/wiki/Known-Issues#click-commands-are-not-executed) where click commands that produce some output are not properly executed.

Can you confirm that redirecting the output to `/dev/null` (`>/dev/null 2>&1`) fixes your issue?

The original issue for this is #916, and @Lomadriel has provided a fix in #1680, I just haven't reviewed it yet.

I will leave this open until you confirm it is the same bug.

**Comment by DrunkenChicken** [comment]
``` 
click-right = chromium-browser >/dev/null 2>&1 &
```
or just 
``` 
click-right = chromium-browser >/dev/null &
```
Both fix my issue, it should be the same bug. 

Sorry for duplicate issue.
